order.go

```diff
@@ -9,37 +9,100 @@ import (
 	"time"
 )
 
-// NewOrder initiates a new order for a new certificate.
+// NewOrder initiates a new order for a new certificate. This method does not
+// use ACME Renewal Info.
 func (c Client) NewOrder(account Account, identifiers []Identifier) (Order, error) {
-	newOrderReq := struct {
-		Identifiers []Identifier `json:"identifiers"`
-	}{
-		Identifiers: identifiers,
-	}
 	newOrderResp := Order{}
-	resp, err := c.post(c.dir.NewOrder, account.URL, account.PrivateKey, newOrderReq, &newOrderResp, http.StatusCreated)
+	err := newOrderResp.postNewOrder(c, account, identifiers, false, "")
 	if err != nil {
 		return newOrderResp, err
 	}
 
-	newOrderResp.URL = resp.Header.Get("Location")
-
 	return newOrderResp, nil
 }
 
-// NewOrderDomains is a wrapper for NewOrder(AcmeAccount, []AcmeIdentifiers)
-// Creates a dns identifier for each provided domain
+// NewOrderDomains is a wrapper for NewOrder(AcmeAccount, []AcmeIdentifiers). It
+// creates a dns identifier for each provided domain. This method does not use
+// ACME Renewal Info.
 func (c Client) NewOrderDomains(account Account, domains ...string) (Order, error) {
+	ids, err := domainsToIds(domains)
+	if err != nil {
+		return Order{}, err
+	}
+
+	return c.NewOrder(account, ids)
+}
+
+// NewOrderRenewal takes an existing *x509.Certificate and initiates a new order
+// for a new certificate, but with the order being marked as a replacement.
+// Replacement orders are exempt from Let's Encrypt NewOrder rate limits. It
+// creates a dns identifier for each provided domain. At least one identifier
+// must match the list of identifiers from the parent order to be considered as
+// a valid replacment order.
+// See https://datatracker.ietf.org/doc/html/draft-ietf-acme-ari-03#section-5
+func (c Client) NewOrderRenewal(account Account, oldCert *x509.Certificate, domains ...string) (Order, error) {
```

Comment:
Is there a compelling reason to make this a separate method, rather than adding a new `replaces *Order` argument to the existing c.NewOrderDomains()?

Reply:
I didn't want to affect other downstream code that calls `c.NewOrderDomains`. I can change it if you want.